File: 2022-convert-1d-array-into-2d-array/2022-convert-1d-array-into-2d-array.py
class Solution:
    def construct2DArray(self, original: List[int], m: int, n: int) -> List[List[int]]:
        if len(original) != m * n:
            return []

        result = []
        for i in range(m):
            row = original[i * n:(i + 1) * n]
            result.append(row)

        return result

File: 2022-convert-1d-array-into-2d-array/2022-convert-1d-array-into-2d-array.cpp
class Solution {
public:
    vector<vector<int>> construct2DArray(vector<int>& original, int m, int n) {
        if (original.size() != m * n) {
        return {};
    }

    vector<vector<int>> result(m, vector<int>(n));
    for (int i = 0; i < m; ++i) {
        for (int j = 0; j < n; ++j) {
            result[i][j] = original[i * n + j];
        }
    }

    return result;
    }
};